mono/utils/mono-state.c

```diff
@@ -236,6 +240,62 @@ mono_summarize_timeline_phase_log (MonoSummaryStage next)
 	return;
 }
 
+static void
+mem_file_name (long tag, char *name, size_t limit)
+{
+	name [0] = '\0';
+	pid_t pid = getpid ();
+	g_snprintf (name, limit, "mono_crash.mem.%d.%lx.blob", pid, tag);
+}
+
+gboolean
+mono_state_alloc_mem (MonoStateMem *mem, long tag, size_t size)
```

Comment:
How is it different and/or better than the existing lock free allocator used by sgen? It's available in `lock-free-alloc.c`? It's already async safe.

Reply:
This is the same reason that I had to stop using certain APIs from the thread dumper. `lock-free-alloc.c` uses hazard pointers in ways that causes crashes early in the initialization to not work. This impacted crashes around coop / hybrid-suspend work in particular. Rather than changing the lock free allocator and slowing down the other uses, it was easier to use a simple file-based allocator.

Reply:
In the case that mono crashes while dumping, these also act as micro-core-dumps.